## Project Details
For this project, we are using the Zomato API.
Zomato APIs give you access to the freshest and most exhaustive information for over 1.5 million restaurants across 10,000 cities globally.

Zomato API documentation link-https://developers.zomato.com/api

### Our Motivation for using Zomato API:
By Collecting the data using Zomato API one can recommend restaurants on the basis of user’s affinity to specific cuisines, establishment types, locations, and price bands.
We can find out whether restaurant support online reservation or not.
We can find what is the most popular and/ or exclusive/new at a given location & time.


## Zomato cuisine problem details
Mexican cuisine began about 9,000 years ago, when agricultural communities such as the Maya formed, domesticating maize, creating the standard process of corn nixtamalization, and establishing their foodways.
You are a huge fan of ‘Mexican’ cuisine. Now using the Zomato API you want to analyse the best restaurant serving the Mexican Cuisine, their locations, and cost for two, etc.
There is a list of questions related to Mexican cuisines given in the next section.


## Zomato Cuisine 1.1
Fetch the cuisine_id of ‘Mexican’ cuisine using Zomato API.
Print the cuisine_id.

Output Format:
cuisine_id

In [33]:
import requests
import json
# API Key: 8ed094119a0b86b85f48073fe55974cc
header={}
p=requests.get('https://developers.zomato.com/api/v2.1/cuisines',params={'city_id':1},headers={'user-key': '8ed094119a0b86b85f48073fe55974cc'})
python_data=p.json()
#print(python_data)
for i in python_data['cuisines']:
    if i['cuisine']['cuisine_name']=='Mexican':
        print(i['cuisine']['cuisine_id'])

73


## Zomato Cuisine 1.3
Fetch the entityid and entitytype of place Connaught Place.
Print the entityid and entitytype.

Output Format:
entity_type entity_id

In [35]:
import requests
import json
# API Key: 8ed094119a0b86b85f48073fe55974cc
header={}
p=requests.get('https://developers.zomato.com/api/v2.1/locations?query=Connaught%20Place',headers={'user-key': '8ed094119a0b86b85f48073fe55974cc'},
                                                                                                 params={'query':'Connaught Place'})
python_data=p.json()
#print(python_data)
for i in python_data['location_suggestions']:
    print(i['entity_type'],i['entity_id'])

{'location_suggestions': [{'entity_type': 'subzone', 'entity_id': 104, 'title': 'Connaught Place, New Delhi', 'latitude': 28.632832, 'longitude': 77.219688, 'city_id': 1, 'city_name': 'Delhi NCR', 'country_id': 1, 'country_name': 'India'}], 'status': 'success', 'has_more': 0, 'has_total': 0, 'user_has_addresses': True}
subzone 104


## Zomato Cuisine 1.4
Fetch the top 10 best-rated restaurant serving ‘Mexican’ cuisine present in Connaught Place.
Print the restaurant name, user rating and restaurant id.

Output Format:

name_1 rating_1 id_1

name_2 rating_2 id_2

name_3 rating_3 id_3

. . .

. . .


In [55]:
import requests
import json
# API Key: 8ed094119a0b86b85f48073fe55974cc
p=requests.get('https://developers.zomato.com/api/v2.1/search?entity_id=104&entity_type=subzone&count=10&lat=28.632832&lon=77.219688&cuisines=73&sort=rating',headers={'user-key': '8ed094119a0b86b85f48073fe55974cc'})
python_data=p.json()
#print(python_data)
for i in python_data['restaurants']:
    print(i['restaurant']['name'],i['restaurant']['user_rating']['aggregate_rating'],i['restaurant']['id'])

Farzi Cafe 4.5 18233593
Station Bar 4.5 18851371
Ministry Of Beer 4.4 18418277
Lord Of The Drinks 4.4 312603
Chili's Grill & Bar 4.4 18537536
Local 4.3 18382360
Warehouse Cafe 4.3 301605
Xero Degrees 4.3 18729557
The Junkyard Cafe 4.3 313256
Teddy Boy 4.3 18281985


## Zomato Cuisine 1.5
Fetch the category id of category type ‘Cafes’ using Zomato API.
Print the category_id.

Output Format:
category_id

In [57]:
import requests
import json
# API Key: 8ed094119a0b86b85f48073fe55974cc
p=requests.get('https://developers.zomato.com/api/v2.1/categories',headers={'user-key': '8ed094119a0b86b85f48073fe55974cc'})
python_data=p.json()
print(python_data)
for i in python_data['categories']:
    if i['categories']['name']=='Cafes':
        print(i['categories']['id'])

{'categories': [{'categories': {'id': 1, 'name': 'Delivery'}}, {'categories': {'id': 2, 'name': 'Dine-out'}}, {'categories': {'id': 3, 'name': 'Nightlife'}}, {'categories': {'id': 4, 'name': 'Catching-up'}}, {'categories': {'id': 5, 'name': 'Takeaway'}}, {'categories': {'id': 6, 'name': 'Cafes'}}, {'categories': {'id': 7, 'name': 'Daily Menus'}}, {'categories': {'id': 8, 'name': 'Breakfast'}}, {'categories': {'id': 9, 'name': 'Lunch'}}, {'categories': {'id': 10, 'name': 'Dinner'}}, {'categories': {'id': 11, 'name': 'Pubs & Bars'}}, {'categories': {'id': 13, 'name': 'Pocket Friendly Delivery'}}, {'categories': {'id': 14, 'name': 'Clubs & Lounges'}}]}
6


## Zomato Cuisine 1.6
Fetch the best-rated restaurant for ‘Mexican’ cuisine with category type ‘ Cafes’ present in Connaught Place using Zomato API.
Print the restaurant name, user rating and restaurant id.

Output Format:

name_1 rating_1 id_1

name_2 rating_2 id_2

name_3 rating_3 id_3

. . .

. . .

In [72]:
import requests
import json
# API Key: 8ed094119a0b86b85f48073fe55974cc
p=requests.get('https://developers.zomato.com/api/v2.1/search?entity_id=104&entity_type=subzone&lat=28.632832&lon=77.219688&cuisines=73&category=6&sort=rating&order=desc',
               headers={'user-key': '8ed094119a0b86b85f48073fe55974cc'})
python_data=p.json()
#print(python_data)
for i in python_data['restaurants']:
    print(i['restaurant']['name'],i['restaurant']['user_rating']['aggregate_rating'],i['restaurant']['id'])

Xero Degrees 4.3 18729557
Chaayos - Meri Wali Chai 4.2 18237315
The Art House Bar n Cafe 4.0 18863766
Cafe Coffee Day - The Square 3.5 303034


## Zomato Cuisine 1.7
Fetch the latest review of the best-rated restaurant for ‘Mexican’ cuisine with Category type ‘Cafes’ present in Connaught Place using Zomato API.
Print the name of user ,user rating and review text.

Note :
Due to limitation of basic API plan it, can only fetch 5 latest review

Output Format:

name1 user_rating1 review_text1

name2 user_rating2 review_text2

name3 user_rating3 review_text3

name4 user_rating4 review_text4

name5 user_rating5 review_text5

In [74]:
import requests
import json
# API Key: 8ed094119a0b86b85f48073fe55974cc
p=requests.get('https://developers.zomato.com/api/v2.1/reviews?res_id=18729557',
               headers={'user-key': '8ed094119a0b86b85f48073fe55974cc'})
python_data=p.json()
#print(python_data)
for i in python_data['user_reviews']:
    print(i['review']['user']['name'],i['review']['rating'],i['review']['review_text'])

Ishu 5 A cozy place with lipsmacking food is what depicts Xero Degrees. They've a wide variety of dishes to choose from. I ordered chilly x chicken fries and chicken wings and absolutely fell in love with them. The taste was a 10/10 for me. The decor was beautiful. The service was prompt and the place was sanitised and safe to eat. What more could one ask for? Absolutely enjoyed my time there. Will visit again soon!
Ishika 5 An outlet which can totally be trusted amidst the pandemic. They've maintained all the safety requirements and have taken utmost care with the preparation of their food. Their updated menu has some of the most amazing dishes that you'll ever have. My favourites were the chicken boneless strips and the nutella oreo waffle. The food has both, top notch quality and wholesome quantity. Loved every bit of it!
Bhukkaddelhi Ke 5 
Nadeem Sarwar 5 If CP is famous for best cafe then the Xero Degrees is one of them..  Best quality of cheese 🧀 items and best of pasta 🍝 to eat.

## Zomato restaurant Problem Detail
Problem 2
"Pa Pa Ya" is a modern Asian bistro & tapas bar from the House of Massive Restaurants and best described as a chic, modern and radical reinterpretation of Asian cuisine.
You have heard from your friends that restaurant "Pa Pa Ya" in Delhi is quite good. So now you want to extract the information related to "Pa Pa Ya" using Zomato API.
There is a list of questions related to the restaurant given in the next section.


## Zomato Restaurant 2.2
Fetch the details of "Pa Pa Ya" restaurant using Zomato API.
Print the user rating, the average cost for two, cuisines and address of "Pa Pa Ya" restaurant.

Output Format:

user_rating

average_cost_for_two

cuisines

address

In [83]:
import requests
import json
# API Key: 8ed094119a0b86b85f48073fe55974cc
p=requests.get('https://developers.zomato.com/api/v2.1/search?entity_id=508&entity_type=group&lat=28.5291530566&lon=77.2192720324&sort=rating&order=desc',headers={'user-key': '8ed094119a0b86b85f48073fe55974cc'})
python_data=p.json()
#print(python_data)
#to fetch id of Pa Pa Ya
for i in python_data['restaurants']:
    if i['restaurant']['name']=='Pa Pa Ya':
        print(i['restaurant']['id'])

18429148


In [92]:
import requests
import json
# API Key: 8ed094119a0b86b85f48073fe55974cc
p=requests.get('https://developers.zomato.com/api/v2.1/restaurant?res_id=18429148',headers={'user-key': '8ed094119a0b86b85f48073fe55974cc'})
python_data=p.json()
#print(python_data)
print(python_data['user_rating']['aggregate_rating'])
print(python_data['average_cost_for_two'])
print(python_data['cuisines'])
print(python_data['location']['address'])

4.9
2500
Asian, Chinese, Thai, Japanese
A-3, Select Citywalk, District Centre, Saket, New Delhi


## Zomato Restaurant 2.3
Find out whether online table reservation is supported in "Pa Pa Ya" restaurant or not using Zomato API.
Print 'yes' if online table reservation is supported else 'no'
Note :
In Zomato response JSON. 1=Yes and 0 = No

Output Format:

yes/no

In [94]:
import requests
import json
# API Key: 8ed094119a0b86b85f48073fe55974cc
p=requests.get('https://developers.zomato.com/api/v2.1/restaurant?res_id=18429148',headers={'user-key': '8ed094119a0b86b85f48073fe55974cc'})
python_data=p.json()
#print(python_data)
if python_data['is_table_reservation_supported'] ==1:
    print('yes')
else:
    print('no')

yes


## Zomato Restaurant 2.4
Fetch the latest review of "Pa Pa Ya" restaurant using Zomato API.
Print the name of the user, user rating and review text.
Note :

Due to the limitation of the basic API plan, it can only fetch 5 latest review

Output Format:

name1 user_rating1 review_text1

name2 user_rating2 review_text2

name3 user_rating3 review_text3

name4 user_rating4 review_text4

name5 user_rating5 review_text5

In [95]:
import requests
import json
# API Key: 8ed094119a0b86b85f48073fe55974cc
p=requests.get('https://developers.zomato.com/api/v2.1/reviews?res_id=18429148',headers={'user-key': '8ed094119a0b86b85f48073fe55974cc'})
python_data=p.json()
#print(python_data)
for i in python_data['user_reviews']:
    print(i['review']['user']['name'],i['review']['rating'],i['review']['review_text'])
    

Rahul.bohra3297 5 Visited the restaurant on their first opening day post lockdown. They were following hygiene and sanitization standards very well. All the staff members were reflecting a sense of positivity and it was topped up with excellent professionalism delivered by vikram. And yes, most importantly they have a lovely menu with sumptuous sushi and popular staple main courses with every detail in bringing the flavours. Keep it up.
Abhinav Khanna 5 Watermelon Amuse Bouche: as fancy as it sounds it’s a classic take on a childhood favourite Vietnamese dessert. Although, they don’t mark this dish as dessert. . So, this picture is from my visit to Pa Pa Ya which a modern Asian bistro & tapas bar from the House of Massive Restaurants and best described as a chic, modern and radical reinterpretation of Asian cuisine. Revolutionarising the pan-Asian dining experience, Pa Pa Ya is an assimilation of various cooking techniques and cultures found across the Asian continent. It is as much ab

## Zomato Distance Problem
Distance is also a very important factor while selecting a restaurant. Suppose you are at Coding Ninjas and you want to select the restaurant which is near to you using Zomato API.
Note down the latitude and longitude of Coding Ninjas using Google Maps
There is a list of questions related to restaurant near me given in the next section.

## Zomato distance 3.1
Fetch the cuisine_id of of cuisine ‘BBQ’ using Zomato API.
Print the cuisine_id

Output Format:

cuisine_id

In [97]:
import requests
import json
# API Key: 8ed094119a0b86b85f48073fe55974cc
p=requests.get('https://developers.zomato.com/api/v2.1/cuisines?city_id=1',headers={'user-key': '8ed094119a0b86b85f48073fe55974cc'})
python_data=p.json()
#print(python_data)
for i in python_data['cuisines']:
    if i['cuisine']['cuisine_name']=='BBQ':
        print(i['cuisine']['cuisine_id'])

193


## Zomato distance 3.2
Fetch the list of the top 10 restaurants serving cuisine ‘BBQ’ which are nearest to Coding Ninjas.
Print the restaurant name, user rating, restaurant id and locality in which restaurant is present.

Output Format:

name_1 rating_1 id_1 locality_1

name_2 rating_2 id_2 locality_2

name_3 rating_3 id_3 locality_3

. . .

. . .

In [102]:
#Coding Ninjas coordinates
#latitude-28.6978261 & longitude-77.1405572
#https://developers.zomato.com/api/v2.1/search?entity_id=1&entity_type=city&count=10&lat=28.6978261&lon=77.1405572&cuisines=193&sort=real_distance&order=desc
import requests
import json
# API Key: 8ed094119a0b86b85f48073fe55974cc
p=requests.get('https://developers.zomato.com/api/v2.1/search?entity_id=1&entity_type=city&count=10&lat=28.6978261&lon=77.1405572&cuisines=193&sort=real_distance&order=desc',headers={'user-key': '8ed094119a0b86b85f48073fe55974cc'})
python_data=p.json()
#print(python_data)
for i in python_data['restaurants']:
    print(i['restaurant']['name'],i['restaurant']['user_rating']['aggregate_rating'],i['restaurant']['id'],i['restaurant']['location']['locality'])

Themis Barbecue House 4.3 18204820 Netaji Subhash Place
Barbeque Nation 4.6 9561 Netaji Subhash Place
Amazings Kitchen 0 19370424 Prashant Vihar
Moet's Barbeque 0 19238917 Alipur
UBQ by Barbeque Nation 3.4 18889760 Rohini
Kays Bar-Be-Que 4.0 3079 Ashok Vihar Phase 1
Chicken Adda 0 19179179 Rohini
7teen BBQ Bistro Bnq 3.9 18595610 Moti Nagar
Pishori Chicken 0 19061375 Kirti Nagar
Hunger On Wheels 0 19054336 Vijay Nagar


## Zomato distance 3.3
Fetch the latest review of the restaurant serving cuisine ‘BBQ’ which is most nearest to Coding ninjas?
Print the name of user ,user rating and review text.
Note :
Due to limitation of basic API plan it, can only fetch 5 latest review

Output Format:

name1 user_rating1 review_text1

name2 user_rating2 review_text2

name3 user_rating3 review_text3

name4 user_rating4 review_text4

name5 user_rating5 review_text5


In [103]:
#https://developers.zomato.com/api/v2.1/reviews?res_id=18204820
import requests
import json
# API Key: 8ed094119a0b86b85f48073fe55974cc
p=requests.get('https://developers.zomato.com/api/v2.1/reviews?res_id=18204820',headers={'user-key': '8ed094119a0b86b85f48073fe55974cc'})
python_data=p.json()
for i in python_data['user_reviews']:
    print(i['review']['user']['name'],i['review']['rating'],i['review']['review_text'])

FoodisBae 4 Roof top cafe with good view. The interiors are nice and the dj plays good music. Loved the food they serve here. They have amazing buffet also. Pizza,pasta and drinks were pretty good.service was nice.
Milind 4 If u are bored of Barbeque nation , this place is must try for u🤓🔥😅. It is one of the best barbeque restaurants in Delhi offering rooftop seating . So we opted for rooftop seating for our lunch & the view was amazing 🔥😻. We opted for veg buffet which included a range of starters like crispy corn , paneer tikka , spicy mushroom . Then in main course we tried manchurian with noodles . As a complementary drink I ordered Kiwi mojito which was so refreshing & eye catchy too😻🔥🔥🔥🤓🤓. So if u visit Pitampura next time don't miss this amazing place.👻🔥🤓😻😻
Shashank Agarwal 4 Great place for an experience of grill in north delhi. Great starters, nice main course and average dessert.
Babu Beg 5 Staff was polite (5*) speedy service (5*) food was delicious and had plenty options. T

## Zomato distance 3.4
Experiment with the radius parameter that is present in ‘GET/search’. Fetch the list of top 10 best-restaurant present in a radius of 4 km of coding ninjas.

Print the restaurant name, user rating, restaurant id and locality in which restaurant is present.

Output Format:

name_1 rating_1 id_1 locality_1

name_2 rating_2 id_2 locality_2

name_3 rating_3 id_3 locality_3

. . .

. . .

In [116]:
#https://developers.zomato.com/api/v2.1/search?count=10&lat=28.6978261&lon=77.1405572&radius=4000&sort=rating&order=desc
import requests
import json
# API Key: 8ed094119a0b86b85f48073fe55974cc
p=requests.get('https://developers.zomato.com/api/v2.1/search?count=10&lat=28.6978261&lon=77.1405572&radius=4000&sort=rating&order=desc',headers={'user-key': '8ed094119a0b86b85f48073fe55974cc'})
python_data=p.json()
#print(python_data)
for i in python_data['restaurants']:
    print(i['restaurant']['name'],i['restaurant']['user_rating']['aggregate_rating'],i['restaurant']['id'],i['restaurant']['location']['locality'])

Indian Accent 4.9 2004 The Lodhi, Lodhi Road
The Big Chill 4.9 1614 Khan Market
Cafe Lota 4.9 305525 Pragati Maidan
Pa Pa Ya 4.9 18429148 Select Citywalk Mall, Saket
Kuremal Mohan Lal Kulfi Wale 4.9 303642 Chawri Bazar
Dum-Pukht - ITC Maurya 4.9 2744 ITC Maurya, Chanakyapuri
The Big Chill 4.9 8244 Khan Market
Wenger's 4.8 4249 Connaught Place
Manhattan Bar & Brewery 4.8 303960 Global Foyer Mall, Golf Course Road
Downtown - Diners & Living Beer Cafe 4.8 4959 Sector 29


## Zomato distance 3.5
Compare the location of the restaurant whether really they are in a distance of 4 km to Coding Ninjas. Find out whether these API radius feature is working or not.

Print 'yes' if radius feature work else print 'no'.

Note :
Use google search to find whether restaurant lie in range of 4km or not.

Output Format:

yes/no

### The restaurants listed above are not really in a distance of 4 km to Coding Ninjas. As the diatnce of Indian Accent is more than 20km and that of The Big Chill is 24km. Thus answer is no